hypermap/aggregator/tasks.py

```diff
@@ -81,6 +90,43 @@ def check_layer(self, layer):
         task_error.save()
 
 
+@shared_task(bind=True)
+def index_cached_layers(self):
+    """
+    Index all layers in the Django cache (Index all layers who have been checked).
+    """
+    from hypermap.aggregator.models import Layer
+    from hypermap.aggregator.solr import SolrHypermap
+    from hypermap.aggregator.models import TaskError
+    solrobject = SolrHypermap()
+    layers_cache = cache.get('layers')
+    layers_list = list(layers_cache)
+    print 'There are %s layers in cache: %s' % (len(layers_list), layers_list)
+    batch_size = settings.SEARCH_BATCH_SIZE
+    batch_lists = [layers_list[i:i+batch_size] for i in range(0, len(layers_list), batch_size)]
+    for batch_list_ids in batch_lists:
+        layers = Layer.objects.filter(id__in=batch_list_ids)
+        if batch_size > len(layers):
+            batch_size = len(layers)
+        print 'Syncing %s/%s layers to Solr: %s' % (batch_size, len(layers_cache), layers)
+        try:
+            success, message = solrobject.layers_to_solr(layers)
+            if success:
+                # remove layers from cache here
+                layers_cache = layers_cache.difference(set(batch_list_ids))
+                cache.set('layers', layers_cache)
+            else:
+                task_error = TaskError(
+                    task_name=self.name,
+                    args=batch_list_ids,
+                    message=message
+                )
+                task_error.save()
+        except:
+            print 'There was an exception here!'
+            self.retry()
```

Comment:
"Task was retried for 3 times, which is the celery default" according to this: http://agiliq.com/blog/2015/08/retrying-celery-failed-tasks/
Anyway, we can safely remove the retry line, as layers id causing the problem will be still in the cache, and they will be indexed on the next task run. Thanks looking at the code!